src/Engine/GameEngine.cs

```diff
@@ -29,6 +29,8 @@ public GameEngine(IOptions<TimeOptions<TTime>> config, ITranslateTime<TTime> tim
 
         public void OnNext<T>(T message, EngineTime time) where T : IEvent
         {
+            if (_disposedValue) throw new ObjectDisposedException("GameEngine");
```

Comment:
`"GameEngine"` can probably be `nameof(GameEngine)` to give us easier refactoring?